crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -205,14 +205,31 @@ where
                 self.client().fill_block_env_with_header(&mut block_env, &pending.header)?;
                 self.client().fill_cfg_env_with_header(&mut cfg, &pending.header)?;
                 return Ok((cfg, block_env, pending.hash.into()))
+            } else {
+                // No pending block, use latest
+                at = BlockId::Number(BlockNumberOrTag::Latest);
+                // Update the latest values for the basefee, timestamp and number
+                let block_hash = self
+                    .client()
+                    .block_hash_for_id(at)?
+                    .ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+                let block =
+                    self.block_by_id(at).await?.ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+                let (cfg, mut env) = self.cache().get_evm_env(block_hash).await?;
+                env.basefee = U256::from(block.header.next_block_base_fee().unwrap());
```

Comment:
this should not happen past merge, but lets only update the value if it returns Some to get rid of the unwrap